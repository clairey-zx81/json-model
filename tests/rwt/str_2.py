# generated from model: ./str_2.model.json
from typing import Callable
import re as re

type Jsonable = None|bool|int|float|str|list[Jsonable]|dict[str, Jsonable]
type CheckFun = Callable[[Jsonable, str], bool]
type PropMap = dict[str, CheckFun]
type TagMap = dict[None|bool|float|int|str, CheckFun]


# define "check_model" ('$')
def check_model(value: Jsonable, path: str = "$") -> bool:
    # '$'
    result = isinstance(value, str)
    return result

