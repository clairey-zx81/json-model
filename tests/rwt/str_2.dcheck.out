str_2.0.true.json: True
str_2.0.false.json: False [('', '$', 'expecting a string')]
str_2.1.false.json: False [('', '$', 'expecting a string')]
str_2.2.false.json: False [('', '$', 'expecting a string')]
str_2.3.false.json: False [('', '$', 'expecting a string')]
str_2.4.false.json: False [('', '$', 'expecting a string')]
